pkg/oci/build.go

```diff
@@ -263,7 +271,7 @@ func createImageIndex(ctx context.Context, target oras.Target, o *BuildGadgetIma
 	layers := []ocispec.Descriptor{}
 
 	for arch, path := range o.EBPFObjectPaths {
-		manifestDesc, err := createManifestForTarget(ctx, target, o.MetadataPath, path, o.WasmObjectPath, arch)
+		manifestDesc, err := createManifestForTarget(ctx, target, o.MetadataPath, path, o.WasmObjectPath, arch, o.CreatedDate)
```

Comment:
> This patch ensures that the annotation is set with the same value for both amd64 and arm64.
>
I tested the `main` branch and the `org.opencontainers.image.created` annotation is already the same:

```bash
$ IG_EXPERIMENTAL=true go run --exec "sudo -E" ./cmd/ig/... image build -t joseregistry.azurecr.io/mytest:latest gadgets/trace
_dns/
INFO[0000] Experimental features enabled
Successfully built joseregistry.azurecr.io/mytest:latest@sha256:29500405310af1d442c6c6a282c88d401187d62d9381a168c5e0af1478eafed8

$ IG_EXPERIMENTAL=true go run --exec "sudo -E" ./cmd/ig/... image push joseregistry.azurecr.io/mytest:latest
INFO[0000] Experimental features enabled
Pushing joseregistry.azurecr.io/mytest:latest...
Successfully pushed joseregistry.azurecr.io/mytest:latest@sha256:29500405310af1d442c6c6a282c88d401187d62d9381a168c5e0af1478eafed8

# Create annotation for the index
$ crane manifest joseregistry.azurecr.io/mytest:latest | jq '.annotations."org.opencontainers.image.created"'
"2024-03-14T13:42:41-05:00"

# Create annotation for amd and arm manifests
$ crane manifest joseregistry.azurecr.io/mytest:latest | jq '.manifests[].annotations."org.opencontainers.image.created"'
"2024-03-14T13:42:41-05:00"
"2024-03-14T13:42:41-05:00"
```

Do you have an example of when they could be different?

Reply:
If your build machine is fast enough, you won't see it.

I took the main branch and added this patch:
```patch
--- a/pkg/oci/build.go
+++ b/pkg/oci/build.go
@@ -141,7 +141,7 @@ func annotationsFromMetadata(metadataBytes []byte) (map[string]string, error) {
        if err := yaml.NewDecoder(bytes.NewReader(metadataBytes)).Decode(&metadata); err != nil {
                return nil, fmt.Errorf("decoding metadata file: %w", err)
        }
-
+       time.Sleep(2 * time.Second)
        // Suggested annotations for the OCI image
        // https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys
        annotations := map[string]string{
```

Then, tested with:
```sh
$ IG_EXPERIMENTAL=true go run --exec "sudo -E" ./cmd/ig/... image build -t albantest.azurecr.io/mytest:latest gadgets/trace_dns/
INFO[0000] Experimental features enabled                
Successfully built albantest.azurecr.io/mytest:latest@sha256:ef0fad91c7ff0fb690667f46b0bd3c29a7bb1b6aad12fe81d3d0f3f15269b9b8
$ IG_EXPERIMENTAL=true go run --exec "sudo -E" ./cmd/ig/... image push albantest.azurecr.io/mytest:latest
INFO[0000] Experimental features enabled                
Pushing albantest.azurecr.io/mytest:latest...
Successfully pushed albantest.azurecr.io/mytest:latest@sha256:ef0fad91c7ff0fb690667f46b0bd3c29a7bb1b6aad12fe81d3d0f3f15269b9b8
$ crane manifest albantest.azurecr.io/mytest:latest | jq '.annotations."org.opencontainers.image.created"'
"2024-03-19T13:11:04+01:00"
$ crane manifest albantest.azurecr.io/mytest:latest | jq '.manifests[].annotations."org.opencontainers.image.created"'
"2024-03-19T13:11:04+01:00"
"2024-03-19T13:11:06+01:00"
$
```

Here the two timestamps are different.

Reply:
Got it!